DatadogRUM/Sources/Instrumentation/WatchdogTerminations/WatchdogTerminationAppStateManager.swift

```diff
@@ -0,0 +1,146 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// Manages the app state changes observed during application lifecycle events such as application start, resume and termination.
+internal final class WatchdogTerminationAppStateManager {
+    var vendorIdProvider: VendorIdProvider
+    let dataStore: RUMDataStore
+    let featureScope: FeatureScope
+    let sysctl: SysctlProviding
+
+    @ReadWriteLock
+    var lastAppState: AppState?
+    @ReadWriteLock
+    var isActive: Bool
+
+    init(
+        dataStore: RUMDataStore,
+        featureScope: FeatureScope,
```

Comment:
suggestion/ `RUMDataStore` dependency is redundant if we already pass `FeatureScope`. 

See: [`featureScope.rumDataStore`](https://github.com/DataDog/dd-sdk-ios/blob/dc16eac3e8cb2644573edb0cadfa052816b1d052/DatadogRUM/Sources/Feature/RUMDataStore.swift#L12-L14).